src/blockdev.rs

```diff
@@ -492,38 +492,65 @@ impl Drop for Mount {
 
 #[derive(Debug)]
 pub struct SavedPartitions {
-    sector_size: Option<u64>,
+    sector_size: u64,
     partitions: Vec<(u32, GPTPartitionEntry)>,
 }
 
 impl SavedPartitions {
     pub fn new(disk: &mut File, filters: &[PartitionFilter]) -> Result<Self> {
-        let mut result = Self {
-            sector_size: None,
-            partitions: Vec::new(),
+        // try reading GPT
+        let maybe_gpt = match GPT::find_from(disk) {
+            Ok(gpt) => Some(gpt),
+            // no GPT on this disk, so no partitions to save
+            Err(gptman::Error::InvalidSignature) => None,
+            Err(e) => return Err(e).chain_err(|| "reading partition table")?,
```

Comment:
Nit: I think you can have just one between `return` and `?`, plus you can avoid re-wrapping the error:
```
e => return e.chain_err(...),
```

(but I haven't tried this)

Reply:
The type of the `Ok()` part is different, so `e => return e...` doesn't work.